### PR TITLE
SystemInformationCard cleanups

### PR DESCRIPTION
**Pull request by martinpitt**
Triggered by PR #19118, trying to clean up the last bits of https://cockpit-logs.us-east-1.linodeobjects.com/pull-19118-20230720-180012-952e0fa8-fedora-38-devel/Coverage/pkg/systemd/overview-cards/systemInformationCard.jsx.gcov.html

### CODE REVIEW

**Review by jelly — APPROVED** [review]
Nice update to 2023!

File: pkg/systemd/overview-cards/systemInformationCard.jsx [hunk 1]
```diff
@@ -79,20 +72,20 @@ export class SystemInfomationCard extends React.Component {
...
 
-                    self.setState({ assetTagText: fields.product_serial || fields.chassis_serial || undefined });
+                    this.setState({ assetTag: fields.product_serial || fields.chassis_serial });
                 })
                 .catch(ex => {
                     // try DeviceTree
-                    machine_info.devicetree_info()
-                            .then(fields => self.setState({ assetTagText: fields.serial, hardwareText: fields.model || undefined }))
+                    machine_info.devicetree_info() // not-covered: coverage only runs on x86_64
+                            .then(fields => this.setState({ assetTag: fields.serial, model: fields.model }))
```
Comment (cockpituous) [inline pkg/systemd/overview-cards/systemInformationCard.jsx:84]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19119-20230721-060543-f5c512c9-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/overview-cards/systemInformationCard.jsx [hunk 2]
```diff
@@ -79,20 +72,20 @@ export class SystemInfomationCard extends React.Component {
...
                             .catch(dmiex => {
-                                console.debug("couldn't read dmi info: " + ex);
+                                console.debug("couldn't read dmi info: " + ex.toString());
```
Comment (cockpituous) [inline pkg/systemd/overview-cards/systemInformationCard.jsx:86]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19119-20230721-060543-f5c512c9-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/systemd/overview-cards/systemInformationCard.jsx [hunk 3]
```diff
@@ -79,20 +72,20 @@ export class SystemInfomationCard extends React.Component {
...
                                 console.debug("couldn't read DeviceTree info: " + dmiex.toString());
-                                self.setState({ assetTagText: undefined, hardwareText: undefined });
+                                this.setState({ assetTag: null, model: null });
```
Comment (cockpituous) [inline pkg/systemd/overview-cards/systemInformationCard.jsx:88]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19119-20230721-060543-f5c512c9-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Comment by martinpitt** [comment]
Apparently I missed a few `// not-covered`s, but alas, it's better, and this is a time sink.

**Pull request merged**